docs/spec/transactions/rum_v3_mark.json

```diff
@@ -0,0 +1,108 @@
+{
+    "$id": "docs/spec/transactions/rum_v3_mark.json",
+    "type": ["object", "null"],
+    "description": "A mark captures the timing in milliseconds of a significant event during the lifetime of a transaction. Every mark is a simple key value pair, where the value has to be a number, and can be set by the user or the agent.",
+    "patternProperties": {
+        "^[^.*\"]*$": {
+            "type": ["number", "null"]
+        }
+    },
+    "properties": {
```

Comment:
These defined properties don't behave as expected, the testdata do not match against them. You can try it out by removing the above `patternProperties` section and sending the testdata - they are invalid. The testdata match the `patternProperties` but not the newly defined attributes.  

Looking at [rum_v3_transactions](https://github.com/elastic/apm-server/blob/7d5cfbf872f6df6decc7ca7979db3104578e99b0/docs/spec/transactions/rum_v3_transaction.json#L79) , `marks` are defined with 
```
               "k": {
                    "type": [
                        "object",
                        "null"
                    ],
                    "description": "A mark captures the timing of a significant event during the lifetime of a transaction. Marks are organized into groups and can be set by the user or the agent.",
                    "patternProperties": {
                        "^[^.*\"]*$": {
                            "$ref": "rum_v3_mark.json"
                        }
                    },
                    "additionalProperties": false
```
Everything defined in `rum_v3_mark.json` is therefore nested inside these properties, which is one nesting level too deep, when specifying the mark grouping (`agent`, `navigation_timing`) here.

The reason it was defined as is, was to match all marks without further specification, allowing the RUM agent devs to just add marks on the fly. The marks are defined as `objects` (first patternProperties definition in the transaction file) containing `numbers` (the second patternProperties in the marks file). 

From the changes you made I assume you want to keep the current behavior but additionally spec out the well known shortened names for easier guidance for RUM agent devs. If the assumption is right, you could solve it with something similar to https://github.com/jalvz/apm-server/compare/rumv3-marks...simitt:json-schema-v3-marks?expand=1

Reply:
Thanks, I noticed something was off indeed, but didn't came as far yesterday. I'll pick the necessary changes from your diff and later create a separate PR for cleaning up the schemas (removing the `allOf`, etc).

Reply:
Pushed a fix. I think we want to allow extra fields bellow `marks` ("k"), `agent` ("a"), and `navigation-timing` ("nt"), but I'll double check.
The fix is incomplete, thou. Next problem is that those fields can be strings too. I am not sure how flexible json-schema's `anyOf` is when it comes to merging rules....

Reply:
The proposed solution did allow for additional attributes nested under `marks`,  `agent` and `navigation-timing`, as long as they were matched by the defined properties or the `patternProperties` and being `numbers`. The `additionalProperties: false` ensured that attributes not matching the defined or pattern properties are rejected.

Transaction marks being allowed to be `string` values is a new requirement then, so far they were restricted to being `numbers`. But that should be easily solvable by adding `string` to the allowed types for all the properties.

Reply:
> The proposed solution did allow for additional attributes nested under marks

ok, i think i didn't manage to get that working locally, will check again tomorrow.

> Transaction marks being allowed to be string values is a new requirement

no, i meant it accepts strings when it shouldn't (so it is wrong)

Reply:
It occurred to me that with the changes for the RUM Intake it might not be a good idea to process _undefined_ marks.
When the RUM agent sends additional marks with shortened keys in future versions, the server will only be able to map to something meaningful in an upcoming version. But older server receiving additional marks will just store them. The additional marks won't be easily readable being indexed with shortened keys, and they also won't be comparable with marks processed by newer servers, as the keys will be different. 
Therefore my question, what is the intention of still processing and indexing unknown marks and wouldn't it make more sense to allow but ignore them?

Reply:
Clarified that with them: allow & ignore.